zarr/storage.py

```diff
@@ -2497,7 +2505,7 @@ def _cache_value(self, key: Path, value):
         value_size = buffer_size(value)
         # check size of the value against max size, as if the value itself exceeds max
         # size then we are never going to cache it
-        if self._max_size is None or value_size <= self._max_size:
+        if self._max_size is None or value_size <= self._max_size:  # type: ignore[redundant-expr]
```

Comment:
`self._max_size` is initialised here:
https://github.com/zarr-developers/zarr-python/blob/f58065b221452acd70235902ad59d920da6fb02f/zarr/storage.py#L2406-L2408
and here:
https://github.com/zarr-developers/zarr-python/blob/f58065b221452acd70235902ad59d920da6fb02f/zarr/storage.py#L2417-L2420

In the first case it shouldn't be `None`, but then typing cannot prevent giving the `max_size` parameter the `None` value. In the second case, no typing is involved.?Not sure why ruff thinks `self._max_size is None` is redundant.